lib/jsonapi/serializer.ex

```diff
@@ -136,6 +133,54 @@ defmodule JSONAPI.Serializer do
     end
   end
 
+  @spec build_relationships(Plug.Conn.t(), tuple(), tuple(), tuple(), list()) :: tuple()
+  def build_relationships(
+        conn,
+        {_parent_view, data, _query_includes, _valid_includes} = parent_info,
+        rel_config,
+        acc,
+        options
+      ) do
+
+    {rewrite_key, data_key, rel_view, _include}
+      = extrapolate_relationship_config(rel_config)
+
+    rel_data = Map.get(data, data_key)
+
+    build_relationships(
+      conn,
+      parent_info,
+      rewrite_key,
+      rel_data,
+      rel_view,
+      acc,
+      options
+    )
+  end
+
+  @doc """
+  Given the relationship config entry provided by a JSONAPI.View, produce
+  the extrapolated config tuple containing:
+    - The name of the relationship to be used when serializing
+    - The key in the data the relationship is found under
+    - The relationship resource's JSONAPI.View module
+    - A boolean for whether the relationship is included by default or not
+  """
+  @spec extrapolate_relationship_config(tuple()) :: {atom(), atom(), module(), boolean()}
+  def extrapolate_relationship_config({rewrite_key, {data_key, view, :include}}) do
```

Comment:
These overloaded functions are more readable than the previous implementation. I like it.